### PR TITLE
Add codespell GitHub actions workflow for spell checking code

### PR DESCRIPTION
**Pull request by matejmatuska**
Use [codespell](https://github.com/codespell-project/actions-codespell) for spell checking code in PRs and pushes to master.

All existing typos are fixed or ignored in the following way:
- Files without any words to check are skipped, as they would just
  create false positives.
- Words like `ro` and `fo` are ignored since they are pretty common in
  the codebase and we don't want to trouble contributors with excluding
  every line containing these from spell checking. It's up to
  maintainers to spot these false negatives.

Jira ref: OAMG-7458

### CODE REVIEW

**Comment by fernflower** [comment]
That's a nice idea! Do you have results of a sample run? Won't it complain about existing spelling errors on files not touched by the PR?

**Comment by matejmatuska** [comment]
@fernflower I have sample run in [my fork]( https://github.com/matejmatuska/leapp-repository/runs/7971474436?check_suite_focus=true).

**Comment by matejmatuska** [comment]
The action is currently blocked by the organization policy, so that needs to be sorted out first.

**Comment by fernflower** [comment]
@matejmatuska Nice, not so many errors as I expected. But still, if you enable it as is (checking all code in master and not only files touched by the PR), it will be failing all the time unless we fix the violations right now (and fixing the violations requires changes to many actors and tests probably, the ones relying on specific output). Do we plan to do this?

**Comment by fernflower** [comment]
@matejmatuska Please comment the linter run in Makefile temporarily (in a separate commit). We will discuss that tomorrow with @pirat89

File: .codespell-exclude-lines
Comment (fernflower) [inline .codespell-exclude-lines:1]: I am afraid that going this way instead of creating an ignore words list will create the need to edit .codespell-exclude-lines from time to time which from the POV of the contributors may be not worth the frustration and extra work :( I'd say codespell is here to help the maintainers and reviers (us), not the contributors, so we shouldn't put that on them.
I'd say we can live with occasional typos like 'ro' but not with crazy-from-the-contributors-pov workflow. How about we implement a more relaxed solution like https://github.com/protocolbuffers/protobuf/actions/runs/277120884/workflow?

Comment (matejmatuska) [reply .codespell-exclude-lines:1]: Wouldn't there will be the need to edit the `ignore_words_list` from time to time? If a contributor opens a PR with a valid word which codespell will recognize as an error, then they would have to adjust that. I created the `codespell-exclude-lines` file using a script, maybe that could be an option.

Comment (fernflower) [reply .codespell-exclude-lines:1]: `ignore_words_list` maintenance is leapp's team responsibility and it should be updated a lot less frequently as this solution will create false negatives for spell checking. Also it's an "add once - stays forever". Like with the `fo` example, you will need one rule to cover all further usages of `fo` as a file object.

`.codespell-exclude-lines` moves the responsibility to the contributor and it's an "add once - will fix only one exact occurence" solution. That's why you had to create 3 rules for using `fo`, and every new patch that has `fo` as a file object will need to add another one.

I am really in favor of cutting the contributors some slack here. But if you are unsure - we could definitely put that to the vote, that's what we have cabal meetings for :)

Comment (matejmatuska) [reply .codespell-exclude-lines:1]: Like with the `fo` example, if `fo` is in `ignore_words_list` and there is a `fo` typo in a PR, then it is back on reviewers to find that typo, which is why the spell checking is being introduced in the first place. 

However I agree that this would probably happen very rarely.

Comment (fernflower) [reply .codespell-exclude-lines:1]: Exactly, it's up to the reviewers - it's our job :) And we should not make it the contributors PITA. 
Also even if we as the reviewers miss it - it won't be the end of the world and nothing will really break.

Don't get me wrong - I absolutely love this patch, coming up with this idea was brilliant and it's a great addition to the existing checks we have. My concerns are only about the perfectness / contribution-friendly tradeoff.

**Comment by fernflower** [comment]
@matejmatuska I think I have figured out a better alternative to commenting out isort check - https://github.com/oamg/leapp-repository/pull/954/
Once this is merged you can rebase your patch and get rid of "[Temporarily comment out linter in Makefile as requested](https://github.com/oamg/leapp-repository/pull/951/commits/da2540e92c98890198e0f4fdcd1075576d8e2a12)" commit.

File: .codespell-exclude-lines
Comment (matejmatuska) [reply .codespell-exclude-lines:1]: I think you are right. It is actually @pirat89's idea, so maybe he could give us his opinion?

**Comment by fernflower** [comment]
PR954 is in, you can rebase now.

File: .codespell-exclude-lines
Comment (matejmatuska) [reply .codespell-exclude-lines:1]: Do you think ignoring the word `ect` because of this single line: `FILES = ['/file1', '/file2', '/dir/ect/ory/file3', '/file4', '/file5']` is a good idea? I agree with the `ro` and `fo` examples, but I am not really sure about this one. I would rather change the path (as it is just a test).

Also similar example on this line: `Flags:                           esan3 zarch stfle msa ldisp eimm dfp edat etf3eh highgprs te vx vxd vxe gs vxe2 vxp sort dflt sie`, the offending word is `sie` which is the only valid occurrence in the entire project.

I think a combination of exclude lines file and `ignore_words`, might be the best solution, what do you think?

Comment (fernflower) [reply .codespell-exclude-lines:1]: Yeah, I think having a combination of both is just perfect.

File: .github/workflows/codespell.yml
Comment (fernflower) [inline .github/workflows/codespell.yml:22]: Do we really need to skip some files? Or is it a temporary measure until you catch all issues and fixup into the proper comments?

Comment (matejmatuska) [reply .github/workflows/codespell.yml:21]: Yeah we do, for example `./repos/system_upgrade/common/actors/storagescanner/tests/files/mounts` contains just mount points and mount options so there aren't really any words which should be checked. They would just create unnecessary false positives.

Comment (fernflower) [reply .github/workflows/codespell.yml:21]: Thanks or clarification!

**Review by fernflower — COMMENTED** [review]
I just love this patch, thank for working on this initiative! 
Please squash the commits and I will approve the pr.

**Review by fernflower — APPROVED** [review]
LGTM, merging, yay!

**Pull request merged**
